src/common/Utilities/Util.cpp

```diff
@@ -116,15 +116,75 @@ std::string secsToTimeString(uint64 timeInSecs, bool shortText, bool hoursOnly)
 
     std::ostringstream ss;
     if (days)
-        ss << days << (shortText ? "d" : " Day(s) ");
+    {
+        ss << days;
+        if (timeFormat == TimeFormat::Numeric)
+            ss << ":";
+        else if (timeFormat == TimeFormat::ShortText)
+            ss << "d";
+        else // if (timeFormat == TimeFormat::FullText)
+        {
+            if (days == 1)
+                ss << " Day ";
+            else
+                ss << " Days ";
+        }
+    }
     if (hours || hoursOnly)
-        ss << hours << (shortText ? "h" : " Hour(s) ");
+    {
+        ss << hours;
+        if (timeFormat == TimeFormat::Numeric)
+            ss << ":";
+        else if (timeFormat == TimeFormat::ShortText)
+            ss << "h";
+        else // if (timeFormat == TimeFormat::FullText)
+        {
+            if (hours <= 1)
+                ss << " Hour ";
+            else
+                ss << " Hours ";
+        }
+    }
     if (!hoursOnly)
     {
         if (minutes)
-            ss << minutes << (shortText ? "m" : " Minute(s) ");
-        if (secs || (!days && !hours && !minutes) )
-            ss << secs << (shortText ? "s" : " Second(s).");
+        {
+            ss << minutes;
+            if (timeFormat == TimeFormat::Numeric)
+                ss << ":";
+            else if (timeFormat == TimeFormat::ShortText)
+                ss << "m";
+            else // if (timeFormat == TimeFormat::FullText)
+            {
+                if (minutes == 1)
+                    ss << " Minute ";
+                else
+                    ss << " Minutes ";
+            }
+        }
+        else
+        {
+            if (timeFormat == TimeFormat::Numeric)
+                ss << "0:";
+        }
+        if (secs || (!days && !hours && !minutes))
+        {
+            ss << secs;
```

Comment:
Will this format seconds value as a single digit when between 0 and 9?

Reply:
Yes, but for restart/shutdown notifications this situation is impossible, that why I did not process this case

Reply:
just use fmt

Reply:
@Warpten not worth using fmt separately for multiple parts of this function (and you can't wrap it all in one big format)